schunk_gripper_driver/schunk_gripper_driver/tests/test_logging.py

```diff
@@ -39,3 +46,183 @@ def test_driver_doesnt_fill_disk_space_by_default(log_monitor, lifecycle_interfa
+    )
+    assert set_params_client.wait_for_service(timeout_sec=2)
+    get_params_client = client.create_client(
+        GetParameters, "/schunk/driver/get_parameters"
+    )
+    assert get_params_client.wait_for_service(timeout_sec=2)
+    future = set_params_client.call_async(
+        SetParameters.Request(
+            parameters=[
+                Parameter(
+                    name="log_level",
+                    value=ParameterValue(
+                        type=ParameterType.PARAMETER_STRING, string_value="DEBUG"
+                    ),
+                )
+            ]
+        )
+    )
+    rclpy.spin_until_future_complete(client, future)
+    future = get_params_client.call_async(GetParameters.Request(names=["log_level"]))
+    rclpy.spin_until_future_complete(client, future)
+    assert future.result().values[0].string_value == "DEBUG"
+    future = set_params_client.call_async(
+        SetParameters.Request(
+            parameters=[
+                Parameter(
+                    name="log_level",
+                    value=ParameterValue(
+                        type=ParameterType.PARAMETER_STRING, string_value="INFO"
+                    ),
+                )
+            ]
+        )
+    )
+    rclpy.spin_until_future_complete(client, future)
+    future = get_params_client.call_async(GetParameters.Request(names=["log_level"]))
+    rclpy.spin_until_future_complete(client, future)
+    assert future.result().values[0].string_value == "INFO"
+
+
+@skip_without_gripper
+def test_driver_rejects_invalid_log_level(driver):
+    client = rclpy.create_node("test_bad_log_level")
+    set_params_client = client.create_client(
+        SetParameters, "/schunk/driver/set_parameters"
+    )
+    assert set_params_client.wait_for_service(timeout_sec=2)
+    get_params_client = client.create_client(
+        GetParameters, "/schunk/driver/get_parameters"
+    )
+    assert get_params_client.wait_for_service(timeout_sec=2)
+
+    # Attempt to set an invalid log level
+    future = set_params_client.call_async(
+        SetParameters.Request(
+            parameters=[
+                Parameter(
+                    name="log_level",
+                    value=ParameterValue(
+                        type=ParameterType.PARAMETER_STRING,
+                        string_value="BAD_LOG_LEVEL",
+                    ),
+                )
+            ]
+        )
+    )
+    rclpy.spin_until_future_complete(client, future, timeout_sec=2)
+    assert future.result().results[0].successful is False
+    # Verify the log level was not set to the invalid value
+    future = get_params_client.call_async(GetParameters.Request(names=["log_level"]))
+    rclpy.spin_until_future_complete(client, future)
+    assert future.result().values[0].string_value != "BAD_LOG_LEVEL"
+
+
+@skip_without_gripper
+def test_driver_logs_correct_level(driver):
+    log_level = []
+
+    driver = Driver("test_driver_logs_correct_level")
```

Comment:
Are you sure you need to start an instance of the driver like this? A running driver is already provided through the `driver` fixture here `def test_driver_logs_correct_level(driver):`. Can you explain that?